### PR TITLE
SearchBox component : Replaced labelText by placeholder. Deprecated l…

### LINKED ISSUE #3977: [SearchBox] should use the prop named placeholder instead of labeltext.
### Bug Report

- __Package version(s)__: 5.47.0
- __Browser and OS versions__: Any

Are you willing to submit a PR to fix? Yes

Requested priority: Low

Products/sites affected: The SearchBox Component

#### Describe the issue:

This doesn't work but compiles without error : 
`<SearchBox placeholder="My PlaceHolder" />`

But this works: 
`<SearchBox labelText="My PlaceHolder" />`

The placeholder is inherited from HTMLInputElement

`export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElement> `

#### Actual behavior:

The placeholder prop is present, but doesn't do anything. The component rely on labelText which is different from what you would expect from an input component.

#### Expected behavior:

The SearchBox Component should use placeholder  prop.

### PR DESCRIPTION
**Pull request by cslecours**
…abelText.

#### Pull request checklist

- [x ] Addresses an existing issue: Fixes #3977
- [x ] Include a change request file using `$ npm run change`

#### Description of changes
SearchBox component : Replaced labelText by placeholder. Deprecated labelText.

#### Focus areas to test

Existing code already using labelText prop shouldn't break.
New component using placeholder prop should work.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 1]
```diff
@@ -78,13 +78,13 @@ export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxStat
...
           className={ classNames.field }
-          placeholder={ labelText }
+          placeholder={ labelText ? labelText : placeholder }
           onChange={ this._onInputChange }
           onInput={ this._onInputChange }
           onKeyDown={ this._onKeyDown }
           value={ value }
           disabled={ this.props.disabled }
-          aria-label={ this.props.ariaLabel ? this.props.ariaLabel : this.props.labelText }
+          aria-label={ this.props.ariaLabel ? this.props.ariaLabel : placeholder }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: What if `labelText` is provided?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 2]
```diff
@@ -78,13 +78,13 @@ export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxStat
         <input
           id={ id }
           className={ classNames.field }
-          placeholder={ labelText }
+          placeholder={ labelText ? labelText : placeholder }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: Move this logic up ie `if (label) placeholder = label`, and this line can just be `placeholder={ placeholder }`  so your change in line 87 won't affect those who pass in `labelText`

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ export interface ISearchBoxState {
 @customizable('SearchBox', ['theme'])
 export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
   public static defaultProps: ISearchBoxProps = {
-    labelText: 'Search',
+    placeholder: 'Search',
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx]: nope. This isn't localized...

File: apps/fabric-website/src/components/IconGrid/IconGrid.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
 
     return (
       <div>
```
Comment (dzearing) [inline apps/fabric-website/src/components/IconGrid/IconGrid.tsx]: Can you please add a change file?

I have removed the hardcoded string and fixed the build error for you.

**Comment by UMAknow-cslecours** [comment]
Jest snapshots are different  (Since the "Search" text is no longer there by default), should I update the test accordingly or the snapshot?

**Comment by cslecours** [comment]
The jest snapshot is different since the default text is no longer present. Should I update the snapshot or the test. What's your policy / point of view on this?

**Comment by dzearing** [comment]
Oh good call; @cslecours it's easy; 

```bash
npm run update-snapshots
```

**Comment by dzearing** [comment]
I will pre-emptively approve, once all the lights are green we can merge.

**Comment by cslecours** [comment]
@dzearing Screener fails because "Search" is no longer present.

**Pull request merged**